### DISCUSSION TITLE
is there a way to generate a separate lock file for one of the packages?

### DISCUSSION THREAD

**Discussion by raviteja83** [lerna/lerna#3342] [Q&A]
I have a monorepo with several packages with the following structure.
- apps
    - app1
    - app2
- packages
     - pkg1
     - pk2
     - pk3
     
I want to update yarn.lock in app1 always. As of now, only the root level lock file gets updated. Is there a way to generate a lockfile for app1 or any of the packages for that matter.

**Comment by fahslaj** [comment]
Hi @raviteja83 - Are you using [Yarn Workspaces](https://classic.yarnpkg.com/lang/en/docs/workspaces/)? If so, there should only be the one lockfile in the root as described [here](https://classic.yarnpkg.com/lang/en/docs/workspaces/#toc-why-would-you-want-to-do-this). Note that this is the default for new Lerna projects.

If you really need to have independent lockfiles, then you would have to disable Yarn workspaces. Check out the [configuration docs](https://lerna.js.org/docs/api-reference/configuration#useworkspaces--packages) for details.

**Reply by raviteja83** [reply]
I have tried that approach. The only problem is yarn.lock of apps does not contain packages versions. Anyway I can achieve that?
The reason for having the packages in yarn.lock is the apps get synced to a public repo where the yarn.lock needs to be updated everytime the sync happens.
